src/oop/user_defined_classes/hard/q2/MainTest.java

```diff
@@ -0,0 +1,117 @@
+package oop.user_defined_classes.hard.q2;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+    private final String testClassName = "DishWasher";
+    private final String intAttributeName1 = "forks";
+    private final String intAttributeName2 = "knives";
+    private final String methodName = "loadDishWasher";
+    private ObjectTest classInstance;
+
+    @BeforeEach
+    public void setUp() {
+        String testClassString = "oop.user_defined_classes.hard.q2." + testClassName;
+        classInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void dishWasherClassHasCorrectAttributes() {
+        String missingAttributeMessage =
+                "The attribute %s could not be found in your class. Please make sure you have added it, it is spelled correctly, and has the correct type";
+        String incorrectModifierMessage = "Your %s attribute in the %s class does not have the correct visibility modifier.";
+        assertTrue(classInstance.hasField(intAttributeName1, int.class),
+                String.format(missingAttributeMessage, intAttributeName1));
+        assertTrue(classInstance.hasModifier(intAttributeName1, "private"),
+                String.format(incorrectModifierMessage, intAttributeName1, testClassName));
+        assertTrue(classInstance.hasField(intAttributeName2, int.class),
+                String.format(missingAttributeMessage, intAttributeName2));
+        assertTrue(classInstance.hasModifier(intAttributeName2, "private"),
+                String.format(incorrectModifierMessage, intAttributeName2, testClassName));
+    }
+
+    @Test
+    public void dishWasherClassHasRequiredConstructor() {
+        Class<?>[] arguments= {
+                int.class, int.class
+        };
+        assertTrue(classInstance.hasConstructor(arguments),
+                String.format("Your %s class is missing a required constructor.", testClassName));
+        assertTrue(classInstance.hasConstructor(arguments, new String[]{"public"}),
```

Comment:
I think it's better if you use hasModifier() here because if the arguments in the constructor are wrong but it has the correct modifier this test would fail and say that the modifier is incorrect.